clap_complete_nushell/tests/completion.rs

```diff
@@ -0,0 +1,328 @@
+use std::fs::File;
+use std::io::Read;
+use std::path::Path;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use nu_cli::NuCompleter;
+use nu_command::create_default_context;
+use nu_parser::parse;
+use nu_protocol::{
+    engine::{EngineState, Stack, StateWorkingSet},
+    Value,
+};
+
+use reedline::{Completer, Suggestion};
+
+const SEP: char = std::path::MAIN_SEPARATOR;
+
+// creates a new engine with the current path into the completions fixtures folder
+pub fn new_engine() -> (PathBuf, EngineState, Stack) {
+    // Target folder inside assets
+    let mut dir = Path::new(env!("CARGO_MANIFEST_DIR")).join("tests");
+    dir.push("snapshots");
+
+    let mut dir_str = dir
+        .clone()
+        .into_os_string()
+        .into_string()
+        .unwrap_or_default();
+    dir_str.push(SEP);
+
+    // Create a new engine with default context
+    let mut engine_state = create_default_context();
+
+    // New stack
+    let mut stack = Stack::new();
+
+    // Add pwd as env var
+    stack.add_env_var(
+        "PWD".to_string(),
+        Value::String {
+            val: dir_str.clone(),
+            span: nu_protocol::Span::new(0, dir_str.len()),
+        },
+    );
+
+    #[cfg(windows)]
+    stack.add_env_var(
+        "Path".to_string(),
+        Value::String {
+            val: "c:\\some\\path;c:\\some\\other\\path".to_string(),
+            span: nu_protocol::Span::new(0, dir_str.len()),
+        },
+    );
+
+    #[cfg(not(windows))]
+    stack.add_env_var(
+        "PATH".to_string(),
+        Value::String {
+            val: "/some/path:/some/other/path".to_string(),
+            span: nu_protocol::Span::new(0, dir_str.len()),
+        },
+    );
+
+    // Merge environment into the permanent state
+    let merge_result = engine_state.merge_env(&mut stack, &dir);
+    assert!(merge_result.is_ok(), "{}", merge_result.unwrap_err());
+
+    (dir, engine_state, stack)
+}
+
+// match a list of suggestions with the expected values
+#[track_caller]
+pub fn match_suggestions(expected: &[&str], suggestions: Vec<Suggestion>) {
+    let expected = expected
+        .iter()
+        .map(|s| (*s).to_owned())
+        .collect::<Vec<String>>();
+    let suggestions = suggestions
+        .into_iter()
+        .map(|s| s.value)
+        .collect::<Vec<String>>();
+    assert_eq!(expected, suggestions);
+}
+
+fn external_completion(file_name: &str) -> NuCompleter {
+    // Create a new engine
+    let (dir, mut engine_state, mut stack) = new_engine();
+
+    let path = dir.join(file_name);
+    let mut buf = Vec::new();
+    let mut file =
+        File::open(&path).unwrap_or_else(|_| panic!("Failed to open {}", path.display()));
+    file.read_to_end(&mut buf)
+        .unwrap_or_else(|_| panic!("Failed to open {}", path.display()));
+
+    let (_, delta) = {
+        let mut working_set = StateWorkingSet::new(&engine_state);
+        let block = parse(&mut working_set, None, &buf, false);
+        assert!(working_set.parse_errors.is_empty());
```

Comment:
See https://github.com/nushell/nushell/commit/1b677f167ead53abf0872453cd6f4517b1b1967d#diff-a1c6589238b4f4d6d32472c8d4545918b470559224930a2611ac5ad6787af878 and https://github.com/nushell/nushell/commit/aded2c1937c93399062456a7ab525a1bb2a3f9c0#diff-a1c6589238b4f4d6d32472c8d4545918b470559224930a2611ac5ad6787af878
```suggestion
        let (block, err) = parse(&mut working_set, None, &buf, false, &[]);
        assert!(err.is_none());
```